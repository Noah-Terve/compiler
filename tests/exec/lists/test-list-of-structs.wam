struct dog {
    int age;
    string name;
};

int main() {
    dog d;
    d.age = 5;
    d.name = "Buddy";
    dog c = #l 10, "spot" #r;

    list @l dog @r l = [c, d ];

    println @l int @r(list_length @l dog @r(l));

    dog f = list_at @l dog @r(l, 1);
    println @l string @r(f.name);

    f.age = 11;
    println @l int @r(f.age);
    println @l int @r(c.age);
}

main();