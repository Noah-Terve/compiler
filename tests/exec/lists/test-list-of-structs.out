2
Buddy
11
10
